File: 1003.check-if-word-is-valid-after-substitutions/solution.cpp
class Solution {
public:
	bool isValid(string S) {
		int a = 0, b = 0, c = 0;
		for (int i = 0; i < S.length(); i++) {
			if (S[i] == 'a') a++;
			else if (S[i] == 'b') b++;
			else c++;
			if (a < b || b < c) return false;
		}
		return a == b && b == c;
	}
};

File: 1003.check-if-word-is-valid-after-substitutions/solution.kt
class Solution {
    fun isValid(s: String): Boolean {
        val stk = StringBuilder()
        for (i in s.indices) {
            val c = s[i]
            stk.append(c)
            if (stk.length >= 3 && stk.substring(stk.length - 3) == "abc")
                stk.delete(stk.length - 3, stk.length)
        }
        return stk.isEmpty()
    }
}
